lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -44,43 +57,82 @@ local function playerIsInBuild( ply )
     return !getPlayerPvpMode( ply )
 end
 
-local function movementMultiplier(weaponCount) 
-    return math.Clamp(1 - (1.9^( weaponCount))/100, 0, 1)
+local minMult = 0.2
+local function movementMultiplier( weaponCount )
+    local N = 1 - (1.9^( weaponCount ))/100
+    return math.Clamp( N, 0.5, 1 )
 end
 
 local function setSpeed(ply, multiplier) 
     -- set speed multiplier, 0 - 1
-    ply:SetRunSpeed(math.Clamp(baseRunSpeed*multiplier,minRunSpeed, baseRunSpeed))
-    ply:SetWalkSpeed(math.Clamp(baseWalkSpeed*multiplier,minWalkSpeed, baseWalkSpeed))
+    ply:SetRunSpeed( math.Clamp( baseRunSpeed*multiplier, minRunSpeed, baseRunSpeed ) )
+    ply:SetWalkSpeed( math.Clamp( baseWalkSpeed*multiplier,minWalkSpeed, baseWalkSpeed ) )
 end
 
-local function adjustMovementSpeed(ply) 
+function adjustMovementSpeed(ply) 
     if playerIsInBuild( ply ) then 
-        setSpeed(ply, 1) 
+        setSpeed( ply, 1 ) 
         return
     end
     
     local weapons = ply:GetWeapons()
     local wepCount = 0
     
     -- count weapons
-    for k, weapon in pairs(weapons) do
+    for _, weapon in pairs( weapons ) do
         if nonEffectedWeapons[weapon:GetClass()] == nil then
-            wepCount =  wepCount+1
+            wepCount = wepCount + 1
         end
     end
     
-    local multiplier = movementMultiplier(wepCount) 
+    local multiplier = movementMultiplier( wepCount ) 
     setSpeed(ply, multiplier)
 end
 
+local function dropWeapon( ply )
+    ply:StripWeapon( ply:GetActiveWeapon():GetClass() )
+end
+
+local function dropAndAdjust( ply, weapon )
+    local class = weapon:GetClass()
+    if undroppableWeapons[class] then 
+        ply:ChatPrint("You are unable to drop \"" .. class .. "\".")
+        return 
+    end
+
+    dropWeapon( ply )
+    adjustMovementSpeed( ply )
+end
+
 -- Hook Functions --
 local function onEquipped( wep, ply )
     if not IsValid( ply ) then return end
     adjustMovementSpeed( ply )
 end
 
+local function onChat( ply, msg )
+    msg = string.lower( msg )
+    if chatDropPhrases[msg] then
+        if not ply:Alive() then return "" end
+        local wep = ply:GetActiveWeapon()
+        if not IsValid( wep ) then return "" end 
+
+        dropAndAdjust( ply, wep )
+        return ""
+    end
+end
+
+-- Concommands --
+concommand.Add("dropweapon", function( ply, cmd, args )
```

Comment:
Can we scope this command to CFC? Just in case some other addon tries to make this too.